File: material/samples/NullSafety.java
// program that illustrates null safety in java
public class NullSafety {
    public static void main(String[] args) {
        @NotNull
        String name;
        name = null; //
        String name2;
        name2 = null;
        System.out.println(name2.length()); // compile error
        System.out.println(name2 == null ? 0 : name2.length());
    }

    public static int getLength2(String name) {
        if (name == null)
            return 0;
        return name.length();
    }
}

File: material/samples/NullSafety.kt
fun main() {
    var name: String
    name = null // compile error
    val name2: String?
    name2 = null
    println(name2.length) // compile error
    print(name?.length ?: 0)
}

fun getLength2(name: String?): Int {
    if (name == null) return 0
    return name.length
}
